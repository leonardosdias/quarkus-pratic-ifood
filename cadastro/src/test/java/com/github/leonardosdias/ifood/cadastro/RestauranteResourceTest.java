package com.github.leonardosdias.ifood.cadastro;

// import static io.restassured.RestAssured.given;

// import org.approvaltests.JsonApprovals;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.RestAssured;
import io.restassured.http.ContentType;
import io.restassured.specification.RequestSpecification;
// import jakarta.ws.rs.core.Response.Status;
import org.junit.*;

@QuarkusTest
@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTestResource(CadastroTestLifecycleManager.class)
public class RestauranteResourceTest {

    @Test
    @DataSet("restaurantes-cenario-1.yml")
    public void testBuscarRestaurantes() {
        String resultado = given()
                .when().get("/restaurantes")
                .then()
                .statusCode(200)
                .extract().asString();
        // JsonApprovals.verifyJson(resultado);
    }

    private RequestSpecification given() {
        return RestAssured.given().contentType(ContentType.JSON);
    }

    @Test
    @DataSet("restaurantes-cenario-1.yml")
    public void testAlterarRestaurante() {
        Restaurante dto = new Restaurante();
        dto.nome = "Novo nome";
        Integer parameterValue = 123;
        given()
                .with().pathParam("id", parameterValue)
                .body(dto)
                .when().put("/restaurantes/{id}")
                .then()

                .extract().asString();

        Restaurante findId = Restaurante.findById(parameterValue);

        Assert.assertEquals(parameterValue, findId);
    }

}
