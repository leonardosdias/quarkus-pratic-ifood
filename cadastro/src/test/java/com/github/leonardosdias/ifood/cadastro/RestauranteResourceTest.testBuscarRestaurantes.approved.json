[
    {
        "id": 123,
        "nome": "Novo restaurante",
        "proprietario": "Teste"
    },
    {
        "id": 223,
        "nome": "Novo restaurante",
        "proprietario": "Teste"
    }
]